addons/mail/static/src/model/model_manager.js

```diff
@@ -35,35 +35,45 @@ class ModelManager {
          */
         this.env = env;
         /**
-         * Whether this is currently handling an "update after" on a record.
-         * Useful to determine if we should process computed/related fields.
+         * Various variables that are necessary to handle an update cycle. The
+         * goal of having an update cycle is to delay the execution of computes,
+         * life-cycle hooks and potential UI re-renders until the last possible
+         * moment, for performance reasons.
          */
-        this._isHandlingToUpdateAfters = false;
-        /**
-         * Determine whether an update cycle is currently in progress.
-         * Useful to determine whether an update should initiate an update
-         * cycle or not. An update cycle basically prioritizes processing
-         * of all direct updates (i.e. explicit from `data`) before
-         * processing computes.
-         */
-        this._isInUpdateCycle = false;
-        /**
-         * Fields flagged to call compute during an update cycle.
-         * For instance, when a field with dependents got update, dependent
-         * fields should update themselves by invoking compute at end of
-         * update cycle. Key is of format
-         * <record-local-id><DEPENDENT_INNER_SEPARATOR><fieldName>, and
-         * determine record and field to be computed. Keys are strings because
-         * it must contain only one occurrence of pair record/field, and we want
-         * O(1) reads/writes.
-         */
-        this._toComputeFields = new Map();
-        /**
-         * Map of "update after" on records that have been registered.
-         * These are processed after any explicit update and computed/related
-         * fields.
-         */
-        this._toUpdateAfters = new Map();
+        this._updateCycle = {
```

Comment:
I prefer keeping attributes flat and just prefix with "uc". It becomes too cumbersome when variables are nested in an object.

Reply:
I'm actually considering removing the nested variable but also not adding prefix.

Model manager has pretty much no other purpose than update cycle, and it has been cleaned to a point where all the its attributes are related to it.